src/python/pants/python/python_setup.py

```diff
@@ -70,15 +72,6 @@ def register_options(cls, register):
             help="The parent directory for the requirement resolver cache. "
             "If unspecified, a standard path under the workdir is used.",
         )
-        register(
-            "--resolver-cache-ttl",
-            advanced=True,
-            type=int,
-            metavar="<seconds>",
-            default=10 * 365 * 86400,  # 10 years.
-            help="The time in seconds before we consider re-resolving an open-ended requirement, "
-            'e.g. "flask>=0.2" if a matching distribution is available on disk.',
-        )
```

Comment:
Do we need a deprecation cycle for this, technically? The option would no-op during the transition time. 

Why use it, then? To avoid breaking people whose config has `--resolver-cache-ttl`. But, I'm not strongly convinced by this - it will either either break now or in one major release. Where it's unclear to me is that keeping the option is confusing because it would have no impact anymore.

Reply:
Not even technically, straight up always need to do this and just missed it. Thanks.

Reply:
Fixed.